scripts/table_updates/update_data_table.py

```diff
@@ -423,79 +454,258 @@ def update_redact_table(
     syn.store(Table(full_pt_schema, result, etag=pt_dat_query.etag))
 
 
-def custom_fix_for_cancer_panel_test_table(
+def check_if_all_join_keys_available(
+    left_table: pandas.DataFrame,
+    right_table: pandas.DataFrame,
+    left_on: str,
+    right_on: str,
+    logger: logging.Logger = None,
+):
+    """Check if all join keys are available in the right table
+
+    Args:
+        left_table (pandas.DataFrame): The left table
+        right_table (pandas.DataFrame): The right table
+        left_on (str): The join key of the left table
+        right_on (str): The join key of the right table
+        logger (logging.Logger, optional): The custom logger. Optional. Defaults to None.
+    Returns:
+        logging.Logger: warning message if join key is not available in the right table
+    """
+    missing_join_keys = left_table.loc[
+        ~left_table[left_on].isin(right_table[right_on])
+    ][left_on]
+    if len(missing_join_keys) > 0:
+        logger.warning(
+            f"Missing {', '.join([str(i) for i in missing_join_keys])} in main GENIE table. Please be advised that tier 1a fields for these records will be filled with NaN."
+        )
+
+
+def update_tier1a(
     syn: synapseclient.Synapse,
+    form: str,
     master_table: pandas.DataFrame,
-    logger: logging.Logger,
+    main_genie_table: pandas.DataFrame,
+    column_mapping_table: pandas.DataFrame,
+    bpc_column_list: List[str],
     config: dict,
+    logger: logging.Logger = None,
+    cohort: str = "",
+) -> Tuple[str, pandas.DataFrame]:
+    """Replace tier1a variables in patient_characteristics or cancer_panel_test table with Main GENIE release files
+
+    Args:
+        syn (synapseclient.Synapse): The synapse client connection
+        form (str): The form name, can be either patient_characteristics or cancer_panel_test
+        master_table (pandas.DataFrame): Table of all of the primary or irr BPC tables
+        main_genie_table (pandas.DataFrame): The dataframe of Main GENIE release
+        column_mapping_table (pandas.DataFrame): The column mapping table between BPC and Main GENIE
+        bpc_column_list (List[str]): The column list to be replaced
+        logger (logging.Logger, optional): The custom logger. Optional. Defaults to None.
+        cohort (str, optional): The cohort name. Defaults to "".
+
+    Returns:
+        Tuple[str, pandas.DataFrame]: The synapse ID for BPC table to be modified and the updated table as dataframe
+    """
+    # check the validity of bpc_column_list
+    valid_col = column_mapping_table.loc[
+        column_mapping_table["prissmm_form"] == form,
+    ].prissmm_element.tolist()
+    assert all(
```

Comment:
Nit: assertions are used for tests, I would raise a ValueError here.

Reply:
Okay. I will also tweak old code to remove assertions.